src/main/java/mate/academy/springboot/datajpa/service/ProductService.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.springboot.datajpa.service;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.List;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+
+public interface ProductService {
+    Product save(Product product);
+
+    Product findById(Long id);
+
+    Product update(Product product);
+
+    void delete(Long id);
+
+    List<Product> findAllByPriceBetween(BigDecimal price1, BigDecimal price2);
```

Comment:
```suggestion
    List<Product> findAllByPriceBetween(BigDecimal from, BigDecimal to);
```
Please, don’t  use such naming